src/Symfony/Component/Message/Asynchronous/Middleware/SendMessageMiddleware.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Asynchronous\Middleware;
+
+use Symfony\Component\Message\Asynchronous\Routing\SenderLocatorInterface;
+use Symfony\Component\Message\Asynchronous\Transport\ReceivedMessage;
+use Symfony\Component\Message\MiddlewareInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class SendMessageMiddleware implements MiddlewareInterface
+{
+    private $senderLocator;
+
+    public function __construct(SenderLocatorInterface $senderLocator)
+    {
+        $this->senderLocator = $senderLocator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($message, callable $next)
+    {
+        if ($message instanceof ReceivedMessage) {
+            $message = $message->getMessage();
+        } elseif (!empty($senders = $this->senderLocator->getSendersForMessage($message))) {
+            foreach ($senders as $sender) {
+                $sender->send($message);
+            }
+
+            if (!in_array(null, $senders)) {
```

Comment:
from the docs we want strict comparison here right?

Reply:
Erm... I don't think so, why? :thinking:

Reply:
in general things like https://3v4l.org/LLOtX, assuming the `null` comes from config and can be any value

But does `~` in

```yaml
framework:
    message:
        routing:
            'App\Message\ImportantAction': [App\MessageSender\ImportantActionToEmailSender, ~]
```

actually work? given https://github.com/symfony/symfony/pull/24411/files/56d5da4c0ef5202d277051d502b8d938e132ec23#diff-0e793081ceb720201745c982a568903fR1377 :thinking:

Reply:
Good catch 👍

Reply:
Fixed 😉